src/models/Country.php

```diff
@@ -123,91 +120,155 @@ function($a, $b) {
   public static async function genAllEnabledCountriesForMap(
+    bool $refresh = false,
   ): Awaitable<array<Country>> {
-    return await self::genAll(
-      'SELECT * FROM countries WHERE enabled = 1 ORDER BY CHAR_LENGTH(d)',
-      self::MC_KEY_ALL_ENABLED_COUNTRIES_FOR_MAP,
-    );
+    $mc_result = self::getMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP');
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries = await self::genAll(
+        'SELECT * FROM countries WHERE enabled = 1 ORDER BY CHAR_LENGTH(d)',
+      );
+      self::setMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP', $all_countries);
+      return $all_countries;
+    } else {
+      invariant(
+        is_array($mc_result),
+        'cache return should be an array of Country',
+      );
+      return $mc_result;
+    }
   }
 
   // All enabled and unused countries
   public static async function genAllAvailableCountries(
+    bool $refresh = false,
   ): Awaitable<array<Country>> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT * FROM countries WHERE enabled = 1 AND used = 0',
-    );
-
-    $countries = array();
-    foreach ($result->mapRows() as $row) {
-      $countries[] = await self::countryFromRow($row->toArray());
+    $mc_result = self::getMCRecords('ALL_AVAILABLE_COUNTRIES');
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries = await self::genAll(
+        'SELECT * FROM countries WHERE enabled = 1 AND used = 0',
+      );
+      self::setMCRecords('ALL_AVAILABLE_COUNTRIES', $all_countries);
+      return $all_countries;
+    } else {
+      invariant(
+        is_array($mc_result),
+        'cache return should be an array of Country',
+      );
+      return $mc_result;
     }
-
-    usort(
-      $countries,
-      function($a, $b) {
-        return strcmp($a->name, $b->name);
-      },
-    );
-
-    return $countries;
   }
 
   // Check if country is in an active level
   public static async function genIsActiveLevel(
     int $country_id,
   ): Awaitable<bool> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT COUNT(*) FROM levels WHERE entity_id = %d AND active = 1',
-      $country_id,
-    );
-
-    invariant($result->numRows() === 1, 'Expected exactly one result');
-    return intval(firstx($result->mapRows())['COUNT(*)']) > 0;
+    return Level::genWhoUses($country_id) != null;
   }
 
   // Get a country by id.
-  public static async function gen(int $country_id): Awaitable<Country> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT * FROM countries WHERE id = %d LIMIT 1',
-      $country_id,
-    );
-
-    invariant($result->numRows() === 1, 'Expected exactly one result');
-    $new_country =
-      await self::countryFromRow(firstx($result->mapRows())->toArray());
-    return $new_country;
+  /* HH_IGNORE_ERROR[4110]: Lines #246 and #260 prevent this function from failing to return */
```

Comment:
Do you actually need this or did you just forget to fix this function like the others?